Added CRUD endpoints for automated_emails

cmraible opened the pull request:
ref https://linear.app/ghost/issue/NY-772

This PR adds an API endpoint for the newly created`automated_emails` table. The endpoint will be used by the frontend settings UI to persist the user generated email content for welcome emails. 

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Adds Admin API CRUD endpoints for `automated_emails`, wiring routes, validation, permissions, fixtures, and comprehensive tests.
> 
> - **Admin API**:
>   - **Controller**: New `automated_emails` controller with `browse`, `read`, `add`, `edit`, `destroy` in `ghost/core/core/server/api/endpoints/automated-emails.js`.
>   - **Registration**: Exposed via `endpoints/index.js` as `automatedEmails`.
>   - **Routes**: Added admin routes for `GET/POST/PUT/DELETE /automated_emails` in `web/api/endpoints/admin/routes.js`.
> - **Validation**:
>   - **Input validators**: New `utils/validators/input/automated_emails.js` (name/slug/status/lexical checks) and wired in `utils/validators/input/index.js`.
> - **Permissions & Data**:
>   - **Migration**: Adds CRUD permissions for `automated_email` to `Administrator` and `Admin Integration` (`data/migrations/...add-automated-email-permissions.js`).
>   - **Fixtures**: Insert `automated_email` permissions and role mappings in `data/schema/fixtures/fixtures.json` and test fixtures; updated fixture counts/hashes in related tests.
> - **Tests**:
>   - **E2E**: New `test/e2e-api/admin/automated-emails.test.js` with snapshots.
>   - **Unit/Integration**: Updated fixture counts and integrity hash expectations (`migration.test.js`, `fixture-manager.test.js`, `integrity.test.js`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e2144db2fc30fb9f66b085b00e4c86e09d3d66a9. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

cmraible committed:
Copied database migrations file from notion into docs/

cmraible committed:
Deleted database migrations doc, created database migration skill

cmraible committed:
Added migration to create automated_emails table

cmraible committed:
Added initial draft of add-admin-api-endpoint skill

cmraible committed:
Added detailed reference for API framework to skill

cmraible committed:
Added detailed permissions and validation guides

cmraible committed:
Added CRUD API endpoints for automated_emails resource

ref https://linear.app/ghost/issue/NY-772

Ghost needed admin API endpoints to manage automated emails for features
like member welcome emails. This adds browse, read, add, edit, and
destroy endpoints at /ghost/api/admin/automated_emails with validation
for status, name, and lexical fields. Permissions are restricted to
Administrator and Admin Integration roles only, ensuring proper access
control for this sensitive functionality.

cmraible committed:
Added CRUD API endpoints for automated_emails resource

ref https://linear.app/ghost/issue/NY-772

Ghost needed admin API endpoints to manage automated emails for features
like member welcome emails. This adds browse, read, add, edit, and
destroy endpoints at /ghost/api/admin/automated_emails with validation
for status, name, and lexical fields. Permissions are restricted to
Administrator and Admin Integration roles only, ensuring proper access
control for this sensitive functionality.

cmraible committed:
Fixed exporter test

cmraible committed:
Fixed linter errors

cmraible committed:
Updated migration directory and filename

cmraible committed:
Fixed a typo in .md file

cmraible committed:
Moved validation logic out of the controller query

cmraible committed:
Resolved linter errors

cmraible committed:
Renamed migration file to fix ordering

cmraible commented on ghost/core/core/server/api/endpoints/utils/validators/input/automated_emails.js:
```diff
@@ -0,0 +1,68 @@
+};
+
+const ALLOWED_STATUSES = ['inactive', 'active'];
+const ALLOWED_NAMES = ['member-welcome-email'];
```
Note to self: change this to 'member-welcome-email-free' and 'member-welcome-email-paid' since we're going to split these out

cmraible committed:
Updated validation logic for automated_emails endpoint

cmraible committed:
Reverted incidental change to skill.md

cmraible committed:
Removed duplicated migration

This somehow snuck in in the process of rebasing

cmraible committed:
Updated e2e-api tests for automated_emails

cmraible committed:
Simplified validation for name field

cmraible committed:
Truncated automated_emails table between e2e tests

cmraible committed:
Reorganized e2e tests

cmraible committed:
Fixed example in permissions.md file

troyciesco commented on ghost/core/core/server/api/endpoints/utils/validators/input/automated_emails.js:
```diff
@@ -0,0 +1,67 @@
+const {ValidationError} = require('@tryghost/errors');
+const tpl = require('@tryghost/tpl');
+
+const messages = {
```
we should move this after those ALLOWED constants, then we can do like 
`Status must be one of ${ALLOWED_STATUSES.join(", ")`
and avoid having to update in multiple spots

troyciesco reviewed (approved):
Once that messages object is cleaned up, LGTM 👍

cmraible committed:
Improved validation error messages for better maintainability

cmraible committed:
Cleaned up snapshot files

After iterating on the e2e tests a few times, the snapshots built up
some cruft. This just regenerates them from scratch to clean them up
